plugins/gradle/src/index.ts

```diff
@@ -0,0 +1,131 @@
+import { Auto, IPlugin, execPromise } from '@auto-it/core';
+import { IExtendedCommit } from '@auto-it/core/src/log-parse';
+
+import path from 'path';
+import fs from 'fs-extra';
+import { inc, ReleaseType } from 'semver';
+import { parse } from 'dot-properties';
+
+/** Global functions for usage in module */
+const logPrefix = '[Gradle-Release-Plugin]';
+
+export interface IGradleReleasePluginPluginOptions {
+  /** The file that contains the version string in it. */
+  versionFile?: string;
+
+  /** The command to build the project with */
+  gradleCommand?: string;
+
+  /** A list of gradle command customizations to pass to gradle */
+  gradleOptions?: Array<string>;
+}
+
+/** getPre does this */
+async function getPreviousVersion(path: string): Promise<string> {
+  try {
+    const data = await fs.readFile(path, 'utf-8');
+    const { version } = parse(data);
+
+    if (version) {
+      return version;
+    }
+  } catch (error) {}
+
+  throw new Error('No version was found inside version-file.');
+}
+
+/**  */
+export default class GradleReleasePluginPlugin implements IPlugin {
+  /** The name of the plugin */
+  name = 'Gradle Release Plugin';
+
+  /** The options of the plugin */
+  readonly options: IGradleReleasePluginPluginOptions;
```

Comment:
This will make the `|| ''` below unnecessary.

```suggestion
  readonly options: Required<IGradleReleasePluginPluginOptions>;
```